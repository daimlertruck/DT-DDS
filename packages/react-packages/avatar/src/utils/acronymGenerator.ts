/**
 * @param text String of one or more words
 * @returns {string} of the first letter of each word in uppercase
 */

export const acronymGenerator = (text = ''): string => {
  const textSliced = text.split(' ').slice(0, 2).join(' '); // Get the first two words

  return textSliced
    .split(/\s/)
    .reduce((response, word) => (response += word.slice(0, 1)), '')
    .toUpperCase();
};
