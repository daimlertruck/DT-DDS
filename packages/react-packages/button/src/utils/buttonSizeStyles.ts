import { CustomTheme as Theme } from '@dt-dds/themes';

import { ButtonSize } from '../constants';

export const buttonSizeStyles = (
  size: ButtonSize,
  theme: Theme,
  isOutline: boolean
) =>
  ({
    large: `
      ${theme.fontStyles.buttonLg}
      padding: ${
        isOutline
          ? `calc(${theme.spacing.spacing_50} - 2px)`
          : theme.spacing.spacing_50
      }  ${theme.spacing.s};
    `,
    medium: `
      ${theme.fontStyles.buttonMd}
      padding: ${
        isOutline
          ? `calc(${theme.spacing.spacing_40} - 2px)`
          : theme.spacing.spacing_40
      } ${theme.spacing.xs};
    `,
    small: `
      ${theme.fontStyles.buttonSm}
      padding: ${
        isOutline
          ? `calc(${theme.spacing.spacing_30} - 2px)`
          : theme.spacing.spacing_30
      } ${theme.spacing.spacing_50};
    `,
  }[size]);
