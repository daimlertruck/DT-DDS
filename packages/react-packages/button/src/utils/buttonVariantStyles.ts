import { Theme } from '@dt-ui/react-core';

import { ButtonVariant, ButtonColor, ButtonAppearance } from '../constants';

const getOutlinedButtonStyles = (
  theme: Theme
): Record<ButtonColor, string> => ({
  primary: `
    color: ${theme.palette.primary.default};
    border-color: ${theme.palette.primary.default};
    background-color: transparent;

    &:hover {
      color: ${theme.palette.primary.dark};
      border-color: ${theme.palette.primary.dark};
    }

    &:active {
      color: ${theme.palette.primary.dark};
      border-color: ${theme.palette.primary.dark};
    }

    &:disabled {
      cursor: not-allowed;
      color: ${theme.palette.primary.medium};
      border-color: ${theme.palette.primary.medium};
    }
  `,
  error: `
    color: ${theme.palette.error.default};
    background-color: transparent;
    border-color: ${theme.palette.error.default};

    &:hover {
      color: ${theme.palette.error.dark};
      border-color: ${theme.palette.error.dark};
    }

    &:disabled {
      cursor: not-allowed;
      color: ${theme.palette.error.medium};
      border-color: ${theme.palette.error.medium};
    }
  `,
  secondary: `
    color: ${theme.palette.secondary.default};
    border-color: ${theme.palette.secondary.default};
    background-color: transparent;
    
    &:hover {
      color: ${theme.palette.secondary.dark};
      border-color: ${theme.palette.secondary.dark};
    }
    &:active {
      color: ${theme.palette.secondary.dark};
      border-color: ${theme.palette.secondary.dark};
    }
    &:disabled {
      cursor: not-allowed;
      color: ${theme.palette.secondary.medium};
      border-color: ${theme.palette.secondary.medium};
    }
  `,
});

export const buttonVariantStyles = (
  variant: ButtonVariant,
  theme: Theme,
  color: ButtonAppearance
) => {
  if (color === 'contrast') {
    return {
      solid: `
        color: ${theme.palette.content.heading};
        background-color: ${theme.palette.base.default};
        border-color: ${theme.palette.border.contrast};

        &:hover {
          background-color: ${theme.palette.primary.medium};
          border-color: ${theme.palette.primary.medium};
        }

        &:disabled {
          cursor: not-allowed;
          color: ${theme.palette.content.disabled};
          border-color: ${theme.palette.base.default};
          background-color: ${theme.palette.base.default};
        }
      `,
      outlined: `    
        color: ${theme.palette.content.contrast};
        border-color: ${theme.palette.border.contrast};
        background-color: transparent;

        &:hover {
          color: ${theme.palette.primary.medium};
          border-color: ${theme.palette.primary.medium};
        }

        &:disabled {
          cursor: not-allowed;
          color: ${theme.palette.content.disabled};
          border-color: ${theme.palette.border.default};
        }
      `,
      text: `
        color: ${theme.palette.primary.contrast};
        background-color: transparent;
        
        &:hover {
          color: ${theme.palette.primary.medium};
        }
        
        &:disabled {
          cursor: not-allowed;
          color: ${theme.palette.content.disabled};
        }
      `,
    }[variant];
  } else {
    return {
      solid: `
        border-color: ${theme.palette[color].default};
        color: ${theme.palette[color].contrast};
        background-color: ${theme.palette[color].default};
        
        &:hover {
          color: ${theme.palette[color].contrast};
          background-color: ${theme.palette[color].dark};
          border-color: ${theme.palette[color].dark};
        }

        &:disabled {
          cursor: not-allowed;
          color: ${theme.palette[color].contrast};
          background-color: ${theme.palette[color].medium};
          border-color: ${theme.palette[color].medium};
        }
    `,
      outlined: getOutlinedButtonStyles(theme)[color],
      text: `
        color: ${theme.palette[color].default};
        background-color: transparent;
        
        &:hover {
          color: ${theme.palette[color].dark};
        }
        
        &:disabled {
          cursor: not-allowed;
          color: ${theme.palette[color].medium};
        }
    `,
    }[variant];
  }
};
