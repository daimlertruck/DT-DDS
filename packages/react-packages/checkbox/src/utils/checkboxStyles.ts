import { Theme } from '@dt-ui/react-core';

export const checkboxStyles = (
  theme: Theme,
  isChecked: boolean,
  isDisabled: boolean
) => {
  if (isChecked) {
    return `
        border: none;
        background-color: ${
          isDisabled
            ? theme.palette.primary.light
            : theme.palette.primary.default
        };
    `;
  }
  return ` 
        border: 1px solid ${theme.palette.border.default};
        background-color: ${
          isDisabled ? theme.palette.base.light : theme.palette.base.default
        };
    `;
};
