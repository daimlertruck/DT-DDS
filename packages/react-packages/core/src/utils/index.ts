export { default as debounce } from './debounce';
export { hexToRgba } from './hexToRgba';
export { default as useClickOutside } from './useClickOutside';
export { getTextStyles } from './typography';
export { withProviders } from './testingUtils';
export { renderStatic } from './renderer';
