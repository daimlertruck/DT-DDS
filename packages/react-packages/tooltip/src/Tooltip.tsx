import {
  TooltipContainerProps,
  TooltipContainer,
  TooltipContent,
} from './components';
import { TooltipContextProvider } from './context';

const Tooltip = ({
  children,
  style,
  dataTestId,
  hideDelay,
}: TooltipContainerProps) => (
  <TooltipContextProvider>
    <TooltipContainer
      dataTestId={dataTestId}
      hideDelay={hideDelay}
      style={style}
    >
      {children}
    </TooltipContainer>
  </TooltipContextProvider>
);

Tooltip.Content = TooltipContent;

export default Tooltip;
