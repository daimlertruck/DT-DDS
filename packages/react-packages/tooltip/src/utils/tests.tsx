import { TooltipContextProvider } from '../context';

export const withTooltipProvider = <P extends object>(
  Component: React.ComponentType<P>
) => {
  return ({ ...props }: P) => {
    return (
      <TooltipContextProvider>
        <Component {...(props as P)} />
      </TooltipContextProvider>
    );
  };
};
