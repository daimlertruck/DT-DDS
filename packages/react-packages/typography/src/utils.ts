export const availableElements = [
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'h6',
  'p',
  'span',
  'label',
  'a',
];

export const availableFontStyles = [
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'h6',
  'body1Bold',
  'body1',
  'body2Bold',
  'body2',
  'body3Bold',
  'body3',
  'body4Bold',
  'body4',
  'link1Bold',
  'link1',
  'link2Bold',
  'link2',
  'link3Bold',
  'link3',
  'button1',
  'button2',
  'button3',
  'p',
  'pSmall',
  'pXSmall',
  'pXXSmall',
  'tag',
];

export const availableColors = [
  'black',
  'blue',
  'blue_05',
  'blue_10',
  'blue_20',
  'blue_30',
  'blue_40',
  'blue_50',
  'blue_60',
  'blue_70',
  'blue_80',
  'blue_90',
  'grey_05',
  'grey_10',
  'grey_20',
  'grey_30',
  'grey_40',
  'grey_50',
  'grey_60',
  'grey_70',
  'grey_80',
  'grey_90',
  'grey_100',
  'green',
  'green_05',
  'green_10',
  'green_20',
  'green_30',
  'green_40',
  'green_50',
  'green_60',
  'green_70',
  'green_80',
  'green_90',
  'red',
  'red_05',
  'red_10',
  'red_20',
  'red_30',
  'red_40',
  'red_50',
  'red_60',
  'red_70',
  'red_80',
  'red_90',
  'white',
  'yellow',
  'yellow_05',
  'yellow_10',
  'yellow_20',
  'yellow_30',
  'yellow_40',
  'yellow_50',
  'yellow_60',
  'yellow_70',
  'yellow_80',
  'yellow_90',
  'purple',
  'purple_05',
  'purple_10',
  'purple_20',
  'purple_30',
  'purple_40',
  'purple_50',
  'purple_60',
  'purple_70',
  'purple_80',
  'purple_90',
];
