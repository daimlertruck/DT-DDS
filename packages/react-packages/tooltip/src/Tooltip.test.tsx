import { withProviders } from '@dt-dds/react-core';
import {
  render,
  fireEvent,
  waitForElementToBeRemoved,
} from '@testing-library/react';

import { TooltipContentProps } from './components';
import Tooltip from './Tooltip';
import { withTooltipProvider } from './utils';

describe('<Tooltip /> component', () => {
  const ProvidedTooltip = withProviders(withTooltipProvider(Tooltip));
  it.each`
    direction   | background
    ${'right'}  | ${'opacity'}
    ${'left'}   | ${'opacity'}
    ${'bottom'} | ${'full'}
    ${'top'}    | ${'full'}
  `(
    'renders a tooltip with the direction as $direction and background as $background',
    async ({ direction, background }: TooltipContentProps) => {
      const { getByTestId, findByTestId } = render(
        <ProvidedTooltip>
          Hover me
          <Tooltip.Content background={background} direction={direction}>
            Some content
          </Tooltip.Content>
        </ProvidedTooltip>
      );

      fireEvent.mouseEnter(getByTestId('tooltip-container'));

      expect(await findByTestId('tooltip-content')).toMatchSnapshot();
    }
  );

  it('should hide tooltip after the hide delay', async () => {
    jest.useFakeTimers();

    const { getByTestId, queryByTestId } = render(
      <ProvidedTooltip hideDelay={1000}>
        Hover me
        <Tooltip.Content>Some content</Tooltip.Content>
      </ProvidedTooltip>
    );

    fireEvent.mouseEnter(getByTestId('tooltip-container'));

    fireEvent.mouseLeave(getByTestId('tooltip-container'));

    jest.advanceTimersByTime(1000);

    await waitForElementToBeRemoved(() => getByTestId('tooltip-content'));

    expect(queryByTestId('tooltip-content')).not.toBeInTheDocument();
  });
});
