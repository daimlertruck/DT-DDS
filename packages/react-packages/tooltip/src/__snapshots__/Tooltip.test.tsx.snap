// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> component renders a tooltip with the direction as bottom and background as full 1`] = `
.emotion-0 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: calc(200px + 2 * 8px);
  overflow-wrap: break-word;
  padding: 8px;
  background-color: #292929;
  box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.1),0px 1px 2px -1px rgba(0, 0, 0, 0.1);
  color: #FFFFFF;
  border-radius: 0px;
  text-align: left;
  white-space: initial;
  position: absolute;
  z-index: 11;
  top: 12px;
  left: 0px;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-0:after {
  content: "";
  position: absolute;
  border: 0px solid transparent;
}

.emotion-0:after {
  bottom: 100%;
  left: 50%;
  margin-left: -7px;
  border-width: 0px 7px 8px 7px;
  border-bottom-color: #292929;
}

<div
  class="emotion-0 emotion-1"
  data-testid="tooltip-content"
  direction="bottom"
  role="tooltip"
>
  Some content
</div>
`;

exports[`<Tooltip /> component renders a tooltip with the direction as left and background as opacity 1`] = `
.emotion-0 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: calc(200px + 2 * 8px);
  overflow-wrap: break-word;
  padding: 8px;
  background-color: #292929;
  box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.1),0px 1px 2px -1px rgba(0, 0, 0, 0.1);
  color: #FFFFFF;
  border-radius: 0px;
  text-align: left;
  white-space: initial;
  position: absolute;
  z-index: 11;
  top: 0px;
  left: 0px;
  opacity: 80%;
  -webkit-transform: translate(-100%, -50%);
  -moz-transform: translate(-100%, -50%);
  -ms-transform: translate(-100%, -50%);
  transform: translate(-100%, -50%);
}

.emotion-0:after {
  content: "";
  position: absolute;
  border: 0px solid transparent;
}

.emotion-0:after {
  top: 50%;
  left: 100%;
  margin-top: -7px;
  border-width: 7px 0px 7px 8px;
  border-left-color: #292929;
}

<div
  class="emotion-0 emotion-1"
  data-testid="tooltip-content"
  direction="left"
  role="tooltip"
>
  Some content
</div>
`;

exports[`<Tooltip /> component renders a tooltip with the direction as right and background as opacity 1`] = `
.emotion-0 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: calc(200px + 2 * 8px);
  overflow-wrap: break-word;
  padding: 8px;
  background-color: #292929;
  box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.1),0px 1px 2px -1px rgba(0, 0, 0, 0.1);
  color: #FFFFFF;
  border-radius: 0px;
  text-align: left;
  white-space: initial;
  position: absolute;
  z-index: 11;
  top: 0px;
  left: 12px;
  opacity: 80%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-0:after {
  content: "";
  position: absolute;
  border: 0px solid transparent;
}

.emotion-0:after {
  top: 50%;
  right: 100%;
  margin-top: -7px;
  border-width: 7px 8px 7px 0px;
  border-right-color: #292929;
}

<div
  class="emotion-0 emotion-1"
  data-testid="tooltip-content"
  direction="right"
  role="tooltip"
>
  Some content
</div>
`;

exports[`<Tooltip /> component renders a tooltip with the direction as top and background as full 1`] = `
.emotion-0 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: calc(200px + 2 * 8px);
  overflow-wrap: break-word;
  padding: 8px;
  background-color: #292929;
  box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.1),0px 1px 2px -1px rgba(0, 0, 0, 0.1);
  color: #FFFFFF;
  border-radius: 0px;
  text-align: left;
  white-space: initial;
  position: absolute;
  z-index: 11;
  top: 0px;
  left: 0px;
  -webkit-transform: translate(-50%, -100%);
  -moz-transform: translate(-50%, -100%);
  -ms-transform: translate(-50%, -100%);
  transform: translate(-50%, -100%);
}

.emotion-0:after {
  content: "";
  position: absolute;
  border: 0px solid transparent;
}

.emotion-0:after {
  top: 100%;
  left: 50%;
  margin-left: -7px;
  border-width: 8px 7px 0px 7px;
  border-top-color: #292929;
}

<div
  class="emotion-0 emotion-1"
  data-testid="tooltip-content"
  direction="top"
  role="tooltip"
>
  Some content
</div>
`;
